GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -846,29 +846,45 @@ private void ShowLoading(bool showSpinner = true)
         }
 
         /// <summary>
-        ///  Indicates whether the revision grid can be refreshed, i.e. it is not currently being refreshed
-        ///  or it is not in a middle of reconfiguration process guarded by <see cref="SuspendRefreshRevisions"/>
+        ///  Indicates whether the revision grid can be refreshed,
+        ///  i.e. it is not in a middle of reconfiguration process guarded by <see cref="SuspendRefreshRevisions"/>
         ///  and <see cref="ResumeRefreshRevisions"/>.
         /// </summary>
-        private bool CanRefresh => !_isRefreshingRevisions && _updatingFilters == 0;
+        private bool CanRefresh => _updatingFilters == 0;
 
         #region PerformRefreshRevisions
 
         /// <summary>
         ///  Queries git for the new set of revisions and refreshes the grid.
         /// </summary>
         /// <exception cref="Exception"></exception>
-        /// <param name="forceRefresh">Refresh may be required as references may be changed.</param>
-        public void PerformRefreshRevisions(Func<RefsFilter, IReadOnlyList<IGitRef>> getRefs = null, bool forceRefresh = false)
+        /// <param name="forceRefreshRefs">Refresh may be required as references may be changed.</param>
+        /// <param name="skipIfAlreadyRefreshing">If true and grid is already / yet being refreshed, do not cancel and restart.</param>
```

Comment:
Where will this be used? another PR?

Reply:
> Where will this be used? another PR?

perhaps in the cases where one can explain why it is really necessary to ignore updates

Reply:
In those situations the guard should be used.